File: read_pdfs.py

import os
import pdfplumber

papers_dir = os.path.join(os.getcwd(), 'papers')
files = [f for f in os.listdir(papers_dir) if f.endswith('.pdf')]

def read_pdf(file_path):
    try:
        with pdfplumber.open(file_path) as pdf:
            text = ""
            for page in pdf.pages:
                text += page.extract_text() + "\n"
            
            output_text = f"\n\n--- START OF {os.path.basename(file_path)} ---\n"
            output_text += text[:50000] # Limit to 50k chars per file
            output_text += f"\n--- END OF {os.path.basename(file_path)} ---\n"
            
            with open('pdfs_extracted.txt', 'a', encoding='utf-8') as f:
                f.write(output_text)
            print(f"Processed {file_path}")
    except Exception as e:
        print(f"Error reading {file_path}: {e}")

if __name__ == "__main__":
    for file in files:
        read_pdf(os.path.join(papers_dir, file))

File: read_pdfs.js

import fs from 'fs';
import * as pdfLib from 'pdf-parse';
import path from 'path';

console.log('pdfLib exports:', Object.keys(pdfLib));
// Try to find the function
const pdf = pdfLib.default || pdfLib;
// If pdf is not a function, we will fail later, but logs will help.

const papersDir = path.join(process.cwd(), 'papers');
const files = fs.readdirSync(papersDir).filter(f => f.endsWith('.pdf'));

async function readPdf(filePath) {
    const dataBuffer = fs.readFileSync(filePath);
    try {
        const data = await pdf(dataBuffer);
        console.log(`\n\n--- START OF ${path.basename(filePath)} ---\n`);
        // Limit text length to avoid overflow? or just print it all and let the agent read it via truncation?
        // 20k chars per file max to start.
        console.log(data.text.slice(0, 20000));
        console.log(`\n--- END OF ${path.basename(filePath)} ---\n`);
    } catch (e) {
        console.error(`Error reading ${filePath}:`, e.message);
    }
}

async function main() {
    for (const file of files) {
        await readPdf(path.join(papersDir, file));
    }
}

main();
